src/ralph/models/edx/navigational.py

```diff
@@ -0,0 +1,130 @@
+"""Navigational event model definitions"""
+from typing import Literal
+
+from pydantic import validator
+
+from ralph.models.selector import selector
+
+from .base import BaseEventFieldModel
+from .browser import BaseBrowserEventModel
+
+
+class PageCloseBrowserEventModel(BaseBrowserEventModel):
+    """Represents the `page_close` browser event.
+
+    This type of event is triggered when the user navigates to the next page
+    or closes the browser window (when the JavaScript `window.onunload` event
+    is called).
+
+    Attributes:
+        name (str): Consists of the value `page_close`
+        event_type (str): Consists of the value `page_close`
+        event (str): Consists of the string value `{}`
+    """
+
+    __selector__ = selector(event_source="browser", event_type="page_close")
+
+    # pylint: disable=unsubscriptable-object
+    name: Literal["page_close"]
+    event_type: Literal["page_close"]
+    event: Literal["{}"]
+
+
+class NavigationalBrowserEventEventFieldModel(BaseEventFieldModel):
+    """Represents the event field of navigational events.
+
+    Note: All navigational events are `browser` events.
+
+    Attributes:
+        id (int): Consists of the edX ID of the sequence.
+        old (str): For `seq_goto`, it consists of the index of the unit being jumped to.
+        For `seq_next` and `seq_prev`, it consists of the index of the unit being navigated to.
+        new (str): For `seq_goto`, it consists of the index of the unit being jumped from.
+        For `seq_next` and `seq_prev`, it consists of the index of the unit being navigated
+        away from.
+    """
+
+    identifier: str # FIXME with string regex
+    new: int
+    old: int
+
+
+class SeqGotoBrowserEventModel(BaseBrowserEventModel):
+    """Represents the `seq_goto` browser event.
+
+    The browser emits such event when a user selects a navigational control.
+    `seq_goto` is emitted when a user jumps between units in a sequence.
+
+    Attributes:
+        event (obj): Consists of member fields that identify specifics triggered event
+        event_type (str): Consists of the value `seq_goto`
+        name (str): Consists of the value `seq_goto`
+    """
+
+    __selector__ = selector(event_source="browser", event_type="seq_goto")
+
+    event: NavigationalBrowserEventEventFieldModel
```

Comment:
Aye, sorry, there is an error in my documentation claiming that the event field is a dict, it's actually a JSON string containing a dict. I'm really sorry for this mistake :disappointed:

Reply:
```suggestion
    event: Union[Json[NavigationalBrowserEventEventFieldModel], NavigationalBrowserEventEventFieldModel]
```
I was hesitating about using `Json` as it changes the output (converts JSON string to dict) but now I think it might be beneficial:
- We can avoid the validation error when validating the event a second time using `Union` (accepting both string + dict)
- It simplifies the conversion step  (we would not have to re-parse the JSON string part)